### PR TITLE
[SPARK-9173][SQL]UnionPushDown should also support Intersect and Except

### PR DESCRIPTION
**Pull request by yjshen**
JIRA: https://issues.apache.org/jira/browse/SPARK-9173

### CODE REVIEW

**Comment by rxin** [comment]
Jenkins, test this please.

cc @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala [hunk 1]
```diff
@@ -84,23 +84,24 @@ object SamplePushDown extends Rule[LogicalPlan] {
 }
 
 /**
- * Pushes operations to either side of a Union.
+ * Pushes operations to either side of a Union, Intersect or Except.
  */
 object UnionPushDown extends Rule[LogicalPlan] {
```
Comment (marmbrus) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala]: `SetOperationPushDown`

**Comment by rxin** [comment]
@yjshen when you rename UnionPushDown, make sure you also rename the test suite, and make the test suite file name matches the class name.

**Comment by yjshen** [comment]
OK, will do soon.

**Comment by rxin** [comment]
LGTM.

**Comment by rxin** [comment]
Thanks - I'm going to merge this.

**Pull request closed without merging**